app/routes/goal_routes.py

```diff
@@ -1 +1,113 @@
-from flask import Blueprint
\ No newline at end of file
+from flask import Blueprint, request, abort, make_response
+from app.models.goal import Goal
+from app.routes.route_utilities import validate_model
+from ..db import db
+from app.models.task import Task
+goals_bp = Blueprint("goals_bp", __name__, url_prefix="/goals")
+
+@goals_bp.post("")
+def create_goal():
+
+    request_body = request.get_json()
+
+    if "title" not in request_body:
+        return {"details": "Invalid data"}, 400
+
+    new_goal = Goal(title=request_body["title"])
+    db.session.add(new_goal)
+    db.session.commit()
+
+    response = {
+        "goal": {
+        "id": new_goal.id,
+        "title": new_goal.title
+    }
+    }
+
+    return response, 201
+
+@goals_bp.get("")
+def get_all_goals():
+
+    query = db.select(Goal)
+
+    goals = db.session.scalars(query)
+    
+    goals_response = []
+    for goal in goals:
+        goals_response.append(
+            {             
+                "id": goal.id,
+                "title": goal.title,
+            }
+        )
```

Comment:
This all looks great! Always feel free to see if you can build this list using comprehension!